### ISSUE TITLE
Fall 2017 Patch

### ISSUE THREAD

**Issue by Azurency** [Azurency/CQUI_Community-Edition#152]
So the Fall 2017 Patch is here. 

We'll keep the good practice to use this issue as the _"master issue"_ of all the fall 2017 patch related issue. ⚠️ **So please report your bugs here and not on separate issues.**

#### 👷‍♀️ About patching CQUI

Yes, we're aware of the issues. It's almost always resolved by merging the patch changes (and there're may be quite a few, we'll see) into the mod files, but often require us to adapt the mod changes to new naming, functions, code removed/moved, etc.
That's why it may take some time (I mean more than a day,  waaay more), if the patch is big or if there're some breaking changes.

#### 🔨 How we'll proceed

We'll try to (roughly) follow these steps, maybe not in that exact order : 
- [x] Merge the changes made in beta branch (new policy reminder, new cqui settings, some cleaning here and there, new readme) into the master branch
- [x] Merge the current pull request (although it's only added files so it should not conflict with the new patch)
- [x] Make the UI files diff of the new patch and have a quick look at them
- [x] Make a list of which files in the mod need to be looked at
- [x] Create a new `Fall-2017` branch
- [x] Patch the changes for CQUI _(the hard work)_
- [x] Look at the integrations of CQUI, maybe they already have patched on their end
- [x] Patch the changes for the integrations _(also not easy)_
- [x] Test the changes
- [x] Merge the `Fall-2017` back into master
- [x] Address the non-resolved issues (if any) that still exist after the patch and that may require more than just merging

#### TL;DR
- Do not create new issue for the new fall 2017 patch
- Post your bugs here
- We're working on it 😉

**Comment by Azurency** [comment]
So the diffs are here : https://github.com/Azurency/Civ6-UIFiles/commit/ab06504988faf39a0601190a87cecf223e895ff2

It's quite a big patch, **221 files** changed (6 822+, 5 945-). More file changed than in summer patch actually but less lines added.

Now that we have the diffs, I'll soon make a list of which files in the mod need to be looked at and changed (and how much it changed, which generally means how much work needs to be done).

**Comment by N7ZebraCakes** [comment]
Was testing civic tree stuff. Probs doesn't help anything, but I get some era-relevant random selections anytime I load into a game in-progress, and can choose one or two before the tree blanks out again.

**Comment by GrantSP** [comment]
- The top panel bugs out by flashing changes several times a second. (Same as the "jiggling issue") 
- City banners are not visible.

**Comment by MauiSven** [comment]
![fehlerliste cqui fall patch 2017](https://user-images.githubusercontent.com/31672167/31851532-e8476586-b667-11e7-9512-7c1ca66c9e76.jpg)

**Comment by N7ZebraCakes** [comment]
@MauiSven - that seems to be a conflict with another mod, as I have CQUI disabled, and have that happen anyway

**Comment by Wiggleworts** [comment]
Hey, this is where we leave bugs, right?  I can't seem to activate projects using the CQUI, they don't show up below wonders.  I've tried reinstalling the mod, and using a couple of different versions.

**Comment by Azurency** [comment]
So, everything from the beta branch and the pull request is merged.
I'm ready to start looking at the changes and make a list of what need to be changed and how much work it'll approximately require.

**Comment by Azurency** [comment]
Here's a list of files that needs to be changed : 

#### CQUI changes
- [ ] `actionpanel.lua`, variable name changed _- easy_
- [ ] `civicchooser.lua`, few additions, extra icons, tabs _- easy_
- [ ] `citysupport.lua`, one line _- easy_
- [ ] `civ6common.lua`, few new lines _- easy_
- [ ] `civilopediasupport.lua`, few changes _- easy_
- [ ] `diplomacyactionview.lua`, lot of changes, need to adapt the cqui modifications _- require work_
- [ ] `diplomacyactionview.lua`, lot of changes but few CQUI modification _- easy_
- [ ] `diplomacydealview.lua`, many changes _- require work_
- [ ] `diplomacydealview.xml`, some changes, need some adaptations _- require work_
- [ ] `diplomacyribbon.lua`, few changes _- easy_
- [ ] `diplomacyribbon.xml`, few changes _- easy_
- [ ] `ingame.lua`, three lines _- easy_
- [ ] `leadericon.lua`, if added, folder changed _- easy_
- [ ] `launchbar.lua`, a lot of changes _- require some work_
- [ ] `launchbar.xml`, a lot of changes _- require some work_
- [ ] `citypanel.lua`, some changes, require to adapt CQUI changes _- require work_
- [ ] `citypanel.xml`, few changes _- easy_
- [ ] `citypaneloverview.lua`, some changes _- require work_
- [ ] `citypaneloverview.lua`, some changes, need to adapt CQUI _- require work_
- [ ] `notificationpanel.lua`, a lot of changes _- require work_
- [ ] `notificationpanel.xml`, few changes _- easy_
- [ ] `productionpannel.lua`, a lot of additions _- require work_
- [ ] `productionpannel.xml`, a lot of tweak _- require some work_
- [ ] `statusmessagepanel.xml`, few modifications _- easy_
- [ ] `unitpanel.lua`, some changes _- require some work_
- [ ] `unitpanel.xml`, few changes _- easy_
- [ ] `partialscreenhooks.lua`, few changes _- easy_
- [ ] `partialscreenhooks.xml`, few changes _- easy_
- [ ] `citystates.lua`, a lot of changes _- require some work_
- [ ] `citystates.xml`, some changes _- require some work_
- [ ] `greatpeoplepopup.lua`, three lines changed _- easy_
- [ ] `greatpeoplepopup.xml`, few changes _- easy_
- [ ] `mappinlistpanel.xml`, two lines _- easy_
- [ ] `mappinpopup.xml`, few lines _- easy_
- [ ] `policyreminderpopup.xml` copied from `techciviccompletedpopup.xml`, autosize changes _- easy_
- [ ] `civicstree.lua`, some changes _- require some work_
- [ ] `governmentscreen.lua`, a lot of changes _- require work_
- [ ] `governmentscreen.xml`, some changes _- require some work_
- [ ] `techtree.lua`, few changes _- easy_
- [ ] `strategicview_mapplacement.lua`, some changes _- require some work_
- [ ] `supportfunctions.lua`, on fonction changed _- easy_
- [ ] `techandcivicsupport.lua`, few changes _- easy_
- [ ] `plottooltip.lua`, some changes _- easy_
- [ ] `toppanel.lua`, a few changes _- easy_
- [ ] `toppanel.xml`, three lines changed _- easy_
- [ ] `unitflagmanager.lua`, new religion flag, attention icon tweak  _- easy_
- [ ] `unitflagmanager.xml`, size change  _- easy_
- [ ] `worldinput.lua`, variable name change, indentation fix, some changes _- require some work_
- [ ] `worldtracker.lua`, some additions _- require some work_
- [ ] `worldtracker.xml`, few changes _- easy_
- [ ] `citybannermanager.lua`, a lot of changes _- require work_
- [ ] `citybannermanager.xml`, religion info added, some changes _- easy_
- [ ] `plotinfo.lua`, few changes _- easy_

#### Integrations changes

BES :
- [ ] `espionagechooser.lua`, if added _- easy_
- [ ] `espionagechooser.xml`, tweaked DestinationInstance and MissionInstance _- easy_
- [ ] `espionageoverview.lua`, added one function, rename on other _- easy_
- [ ] `espionageoverview.xml`, few changes _- easy_

BTS : 
- [ ] `tradeoriginchooser.lua`, two lines _- easy_
- [ ] `traderoutechooser.lua`, many changes _- require work_
- [ ] `traderoutechooser.xml`, some changes that need to be adapted _- require some work_
- [ ] `tradeoverview.lua`, some changes  _- require some work_

ML : 
- [ ] `minimappanel.lua`, some changes _- easy_
- [ ] `minimappanel.xml`, some changes _- easy_
- [ ] `modallenspanel.lua`, new religion lens _- easy_
- [ ] `modallenspanel.xml`, few changes _- easy_

URS : 
- [ ] `reportscreen.lua`, some changes _- require work_
- [ ] `reportscreen.lua`, a few sizes changes _- require some work_

It's a lot of changes everywhere, so it will take some time, at least a week to test everything. But there's also some good news : 

- We can now apparently **extend** the UI files and not replace them completely :tada:, if it works correctly (I needs to test it, and see how it work with xml files) then I will take full advantage of this addition and the next updates should require a lot less work (because they will require a lot less merging)

**Comment by MauiSven** [comment]
Thank you very very much for Information.
Will old savegames work after updating, too?

**Comment by smalltimepro** [comment]
Unable to place either wonder or district. Dont know why that is happening

**Comment by Timmeey86** [comment]
Working the unchecked checkboxes bottom up (skipping mod integrations) in fork #157

**Comment by Azurency** [comment]
That's on way to see it ;)
I'm currently merging city panel related files (there's quite a few) and testing it, I'll move on the diplomacy and notification next I think.

**Comment by oerms** [comment]
Hi. I have edited some of the files and commited to my master. How do you get the changes to your repo?

(sorry, haven't worked much with pull requests or what this is called... :( )

**Comment by jimmyp1979** [comment]
im not and never touch files i appreciate the work gone into it but if steam cant just update the mod without me playing with files that could damage my game then nah unsubscribe to be honest

**Comment by jimmyp1979** [comment]
so we dont have to do anything with our files ourselves?we just wait for a patch? works for some players , quill 18 etc but not for alot or am i missing something(newbie alert sorry)

**Comment by AlexandreDecan** [comment]
No, you don't have to do anything except waiting for an update (unless you want to contribute to the repository and help other maintainers fixing the mod). 

Don't blame the maintainers of CQUI ;-) The fault is Firafix's: they didn't release a list of upcoming changes before the update (nor after, to be honest), meaning that now that the update has been released, the maintainers of this mod have to do their best to take into account the large number of changes Firaxis made to the game.

**Comment by jimmyp1979** [comment]
didnt blame them i was just pointing out that if i had to touch files i would unsubscribe from it , i put i appreciate the work going into it

**Comment by AlexandreDecan** [comment]
@Azurency One thing that can help to prevent people being "frustrated" because of the mod is to display a popup explaining that in its current status, the mod cannot be used to play a game (or contains bugs that could lead to a "less interesting" experience) and to invite them to wait until an update (or to temporarily disable the mod), and to remove this popup when it's no longer relevant.

**Comment by AlexandreDecan** [comment]
Following the idea exposed in my last comment, would it be possible to put an "empty" version of the mod to Steam (with the popup explaining why it is "empty") until the release of a working version? This will (1) allow people to continue their previously saved games, (2) avoid them unsubscribing the mod, (3) warn them an update is coming.

**Comment by Timmeey86** [comment]
I'll work on government screens next, this is gonna take a while

**Comment by smalltimepro** [comment]
@Timmeey86 i have been working on the productionpanel and dependencies but just cant get the district placement to work. Any tips?

**Comment by smalltimepro** [comment]
Found error and fixed. There is an added confirmation dialog for district/wonder placement in strategicview_mapplacement.lua. I have the fixed file ill see if I can find the procedure to push it

**Comment by Azurency** [comment]
@smalltimepro Have you looked at https://github.com/Azurency/CQUI_Community-Edition/issues/152#issuecomment-338484640 ?

**Comment by smalltimepro** [comment]
Yes I have. I have done the strategicview file and requested a pull. Mod is largely working for me of the back of some local changes but this was the crucial one cause it was a bit of a road block.

**Comment by Azurency** [comment]
Well you would have realized that I already merged the files you committed in your pull request

**Comment by Azurency** [comment]
I'm taking some time merging the city-related files because there were some changes in the way we have to check wether or not show a building in the production panel (for exemple the religion building, right now, they all show up in the list)

**Comment by astog** [comment]
@Azurency I'll update the files this weekend, so all my mods should get compatible soon.

**Comment by techgiz** [comment]
When on the deal screen to make a deal with the AI on the AI side of the screen there are not list any items listed of the AI's that can be traded or included in the deal.

**Comment by Azurency** [comment]
**A small update :**
- I'm currently looking at all the pull request for the fall patch (thank you all for the contributions) and I'll merge as much as I can this weekend.
- I'm also going to work on the diplomacy related files (with the new diplomacy screen they added, don't know how I fell about it, but we'll see that later)

With @astog trying to fix its mods during the weekend aswell, I think it's looking great for a patch next week !

**Comment by oerms** [comment]
Please mark `Popups/MapPinPopup.xml` with a check mark. It is already merged.

**Comment by Azurency** [comment]
@oerms Were ? isn't it `mappinlistpanel.xml` and not `mappinpopup.xml` that is already merged ?

**Comment by oerms** [comment]
You are right. Please delete my comment(s),

**Comment by oerms** [comment]
I think everything is being reviewed right now, except the `launchbar.*` files and diplomacy, of course.
Looking very good so far. :)

**Comment by MauiSven** [comment]
@oerms Where or how can we find or get that changes described above?

**Comment by oerms** [comment]
@MauiSven It is still very buggy: Occupation does not end, walls are not repaired, the queue is buggy, you can build all religious buildings, you do not see trade proposals from AI, etc pp...

 I would not recommend using the mod just yet. Wait at least for the merge of existing pull requests. This should be happening this weekend. Then download all files and copy them to your `My Games` folder as usual with mods. Diplomacy might not have been fixed by then, however.

**Comment by MauiSven** [comment]
@oerms Ok, thx for your quick answer.
But I only meant the particial solutions regarding civics tree and wonder/district placement, cause that's it, what I strongly need for my running game. The other tasks are well known but actual not so urgent for me...

**Comment by oerms** [comment]
@MauiSven Get my files and replace this https://github.com/Azurency/CQUI_Community-Edition/compare/fall-2017#diff-be44ff002e3265a8609eaccbf3aecf67
Please understand that we cannot support this further. Direct your disappointment towards Firaxis for not letting the community know about upcoming patches. (And releasing a crappy interface for a great game in the first place...)

**Comment by MauiSven** [comment]
@oerms Thank you and I understood.
But I only found program code behind the link instead completed files, is that correct...?

**Comment by oerms** [comment]
@MauiSven click `view` and `raw` and copy into  `CQUI_Community-Edition/Assets/UI/strategicview_mapplacement.lua`
No further comment from me on this. This is only cluttering our thread here.

**Comment by smalltimepro** [comment]
@MauiSven @oerms You need the /Assets/UI/Panels/productionpanel.lua as well for it to work. ProductionPanel still needs a lot of work. Extra faith buildings show up but not contructable so not a big issue. Wall once damages doesnt give repair option and queue is a bit buggy. Next road block is probably the wall thingie. I made a fresh commit on my fork earlier if you want to pick the latest on those two files

**Comment by Azurency** [comment]
@MauiSven the production queue work, extra faith building don't show up, you can repair walls... Please work on the right branch and stay informed thanks to the comments here on what still need to be done. 
I merged the city related stuff myself (including the production queue, although there may still be some weirdness, we'll see on the testing phase next week)

Also this is not a thread support but an **issue** about the fall patch, no more comment on "how to get the latest version" that does not exist at the moment please.

**Comment by oerms** [comment]
I think I have done the rest of the files which are not connected to the diplomacy and that not integrations can I somehow help with the diplomacy?

**Comment by Timmeey86** [comment]
Combat preview is not shown when right click dragging from an own unit over an enemy unit. It is shown when selecting an own unit and moving the mouse above an enemy unit, however.
Tested on Azurency/fall-2017 with and without changes of #160.

**Comment by Azurency** [comment]
`unitpannel` is still not merged (there's `unitpannel.xml` in #160 and I merged `unitpannel.lua` on my machine and I haven't pushed it yet). So don't worry, I'll tell you when to start the testing phase ;)

**Comment by sparrow8332** [comment]
Productionpannel has a weird refresh bug every time you select an item to be added to production que - districts section looks like it dose a weird expand & refresh then back to how it was.

**Comment by Azurency** [comment]
@sparrow8332 yes, I think it's the new `size="auto, auto"`, it now refresh as each item is added to the stack. I tried to quickly look at it but as it's not game breaking I left it as it is, we'll come back to it when everything is merged and try to find a solution.

**Comment by smithmid** [comment]
@Azurency When do you anticipate pushing what you have?

**Comment by Azurency** [comment]
**Let's make a quick update :**
- All the CQUI files have been merged now, in the `fall-2017` branch
- BES, BTS and ML are being merged by @astog in their respective GitHub repo (I see that he already have pushed an update to BTS)
- IDS is merged

It leaves us with _URS_, I don't think its creator will update it (he have not updated it for the summer patch, or I don't see it on civfanatics. We'll have to update it. I can work on that tomorrow (or if someone want to do that before, fell free to open a pull request before I start 😉)

We'll need to bring back the changes from BES, BTS and ML here (I think @astog will open a pull request, or tell me if you want me to update it here)

Everything looks great for an update on steam next week.

**Comment by oerms** [comment]
How did you do the diff for the files? Let my try and work on the reportscreen. (the second should be xlm, I suppose...)

**Comment by Azurency** [comment]
@oerms Great ! Yes those are the 3 files. The latest version is the one there is in CQUI here.

**Comment by oerms** [comment]
They were not in the fall-2017 branch, were they? I had to copy manually from my old stand-alone plugin.

**Comment by ricanuck** [comment]
@Azurency : URS is at this point VERY different from the original, we added a lot of code and QoL to that file before the Summer fiasco, so I would not want to bring the original author to it as URS is now a completely different (and better) beast.

**Comment by oerms** [comment]
Hi @ricanuck (alias Aristos, I have seen that name in the code ;) )!
I have tried to work the changes of the patch into your modifications. If you can spare the time, please review. Thanks!

**Comment by sparrow8332** [comment]
Using latest fall-2017 branch did a little test game notices NO Unit Report Screen Button.

https://i.imgur.com/ni6I8Kn.jpg

**Comment by MauiSven** [comment]
@sparrow8332:
Yes, indeed, I had already noticed this and reported under the following link:

[Link](https://github.com/Azurency/CQUI_Community-Edition/commit/13d4a2214fb6eff6ff1fa3fb138af3cdf58eeece#comments)

**Comment by Azurency** [comment]
@sparrow8332 @MauiSven again, not everything is merged. The URS is still waiting for review and so the `reportscreen.lua` that adds the icon to the launchbar is not updated.

I will tell you when to start testing and reporting bug that may still exist.

**Comment by JIMTHERATMAN** [comment]
FIX THE FUCKING GAME FFS
STOP SLEEPING AND DO YOUR JOB FOR ONCE PAJEETS

**Comment by Starjon** [comment]
@JIMTHERATMAN You realize that they do that in their spare time and don't get payed for it? They are not Firaxis.

**Comment by JIMTHERATMAN** [comment]
@JIMTHERATMAN You realize that they do that in their spare time and don't get payed for it? They are not Firaxis.
--


@Starjon NO I DID NOT

**Comment by kecker79** [comment]
It's not their job, this is a hobby they do out of the kindness of their hearts in their spare time. You can help the cause by not being an asshole.

**Comment by JIMTHERATMAN** [comment]
Sorry.

**Comment by JIMTHERATMAN** [comment]
Is there an ETA for final release?

**Comment by kecker79** [comment]
If you read back through the history of this effort, it looks like next week. Which given the changes that Firaxis vomited out without warning, I think is a miraculous turnaround.

**Comment by MauiSven** [comment]
Have the content of the 57 changed files (stand now) transmitted and the game checked:
Everything what I couldt check is ok, very good, fine, thank you very much, I'm loving it, I LOVE YOU!
All the best to all devs and contributors!

**Comment by Azurency** [comment]
Ok great, everything is merged now 🎉, we can start testing.

**How to test**
Just pick the source of the `fall-2017` branch, follow the installation instructions and play ! 

I will want to push a steam update Saturday I think, or if everything is ok and I have many feedback, maybe even tomorrow evening. The folks back at the workshop will be happy.

**Comment by BlackSmokeDMax** [comment]
Any preference on completely unsubscribing to other mods as compared to just making sure not to enable other mods? (Besides CQUI itself, I assume it is best to un-subscribe to the Steam version for now)

**Comment by ricanuck** [comment]
For testing with an on-going game that you don't want to lose: add the following to the properties of the modinfo file:

<AffectsSavedGames>0</AffectsSavedGames>

and then load your ongoing game. In fact, we should probably add that line to the modinfo officially, as the mod is pure UI and should not affect savegames anyways.

**Comment by BlackSmokeDMax** [comment]
https://steamuserimages-a.akamaihd.net/ugc/861738022419018962/F889F384BFECA52EFA096AD2E8677057AC296EED/ 

Once I moved the slinger into the city, the builder menu was fully visible once again.

**Comment by MauiSven** [comment]
@ricanuck Do you mean the file "CQUI-SparrowEdit.modinfo" at "..\workshop\content\289070\1125169247"?

**Comment by jelmerj** [comment]
I've been playing a few more turns in a hotseat game I'm playing with a friend of mine and everything seemed to worked perfectly. We couldn't continue previously because my friend wanted to build a wonder and he had to choose a civic, both of which are working now!

**Comment by LordYanaek** [comment]
Setting
`<AffectsSavedGames>0</AffectsSavedGames>`
would probably help get CQUI approved (or at least tolerated) for GotM which would be great. I've been using this trick myself for several games now without any issues. However if i understand correctly CQUI does actually store data inside the saves (options and production queue) so _in a sense_ it does affect saved games.

I currently have a perfectly reproducible issue where the entire UI disappears during AI turn after T77 on the current GotM (using fall-2017 branch). I reloaded the autosave 3 times and got the same issue each time. I'm not sure it's a CQUI bug but i have no other mod. I can tell it's the AI turn because when i bring the menu with the esc key (it's the only part of the UI still working) i can exit but not save. It doesn't seem to actually progress to the point when i once again get my turn (but maybe i didn't wait long enough) which is strange for an UI bug but if it doesn't come from CQUI, then Firaxis seriously messed something in that patch.
I don't know whether i should open a new issue or report in this thread. I can give you all my logs and/or the save if it can help.

**Comment by Timmeey86** [comment]
@LordYanaek I could only test this tomorrow evening (in like 18 hours or something), but if it's reproducible from the save game, at least on your machine, sharing the save game should be the quickest option in order to resolve this.

**Comment by Azurency** [comment]
Yes CQUI store GameConfiguration data for its settings on the save, but you can load CQUI games without the mod, there are several tricks to dot that in fact, and it's not breaking anything. We'll consider adding this.

And yes please, **having the log when reporting any issue is a big help** 👍

**Comment by ricanuck** [comment]
@Azurency : check latest pull req, some important code was missing inside the diplo ribbon lua file, which made the relationship tooltip misbehave. This should go into the Fall update before release.

**Comment by oerms** [comment]
Another bug: When viewing the map through the Religion lens (clicking a Missionary or Apostle) the religious regions and the influence indication disappear when the mouse is over certain icons (I am not sure this is the right term): sometimes cities ("banners"?), sometimes unit icons. But not all cities nor all units but only a few of them.

**Comment by Timmeey86** [comment]
The same bug oerms mentioned happens if you open any Lens in the Lenses menu and move your mouse over a religious unit

**Comment by astog** [comment]
I'll try and fix that today. Thank you for reporting this

**Comment by BlackSmokeDMax** [comment]
So, yesterday I left a report that was a bit mangled, I probably shouldn't have done it in a hurry before I left. Today I'll explain it a bit better. I will also delete my original post.

This issue happened while a builder and slinger were in the same tile. The builder was not able to build a farm because some of the panel containing all of the builder operations was being obscured. I resolved this by moving the slinger out of the tile. I have a screen shot showing this. Note you can see part of the build farm button, but you can not use that button, even in the area where you can fit the mouse pointer. Obviously the ideal behavior is you can still use your builder with a military unit in the same tile, so somewhat of a major issue.

<a href="https://imgur.com/ksBw1hS"><img src="https://i.imgur.com/ksBw1hS.jpg" title="source: imgur.com" /></a>

**Comment by LordYanaek** [comment]
I triple checked the save and got the missing UI again so on my machine at least, it's perfectly reproducible. I let Civ6 run for several minutes just in case and the UI never showed up, nor did the AI turn end.

Here is a zip with the save and my logs. I can't upload it here, i always get a message "Something went really wrong, and we can't process that file. Try again" [so i put it on my dropbox](https://www.dropbox.com/s/ulk3asof3k6dvgh/CQUIFall2017MissingUi.zip?dl=0)

I downloaded the fall-2017 branch yesterday to try it with the current "Game of the month". The mod's folder is called "CQUI_Community-Edition-fall-2017" so i think i didn't mess things up (and i guess i shouldn't have been able to play to T77 with an older version). I have a few mods installed but none currently selected. I have all the DLCs but they were disabled at creation for that save as is usual for GotM. I'm on a PC with Windows 10 in case it's a useful information.

**Comment by Timmeey86** [comment]
@LordYanaek The save game works fine on my machine, but I saw a number of localization warnings in your logs. It looks like the mod fails loading a text for a diplomatic gossip message in your language and eventually throws an exception when trying to prepare that for display.
In short: One possibility is the issue only occurs when using a certain game language. Which language are you using?

**Comment by georgwacker** [comment]
Found a bug where the promotion icon on the unit & the promotion button itself on a scout disappeared. Moving the unit once made it appear again.

**Comment by Azurency** [comment]
@astog Thx, I'll leave it to you then. Open a pull request when it's fixed 👍 

@BlackSmokeDMax Thx for this report, I fixed this bug, the size of the `ExpandSecondaryActionStack` that was auto expanded by CQUI was not updating correctly (I don't get why, it's a new bug introduced with this patch I guess)

@LordYanaek Looks like your translation for `LOC_GOSSIP_SOURCE_DELEGATE` is empty and that made `CQUI_TrimGossipMessage()` in `civ6common.lua` crash. I had this problem myself with other mods that changed some localizations like removing some gossip (Simplified Gossip for instance), but anyway I changed the code so that it would now only print an error and not make the UI disappear or be game breaking (although it's not normal to have this error)

@georgwacker I think this promotion button hiding is related to what I solved for @BlackSmokeDMax but having the logs and maybe a screenshot/savegame will help a lot to identify the issue

**Comment by cpinter** [comment]
I downloaded the fall-2017 branch and started a game. A few issues I noticed until I couldn't continue the game:
* The top bar flickers when end of turn is processing. Science is alright, but culture, faith and gold are resizing and jumping around. It stops when the new turn starts
* When the code of laws civic is done, the governments window pops up for a moment before the policy cards can be chosen
* Cannot open civic tree from any of the buttons. This is a blocker. Not sure what's different on my computer as on others, but here are the logs:
[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1442725/Logs.zip)

Sorry if I'm being redundant, just thought I'd report what I found.

**Comment by TheHandiCraftsman** [comment]
Not sure if it's related to something else, because surely someone else would have reported this by now, but I can't build districts at all.  I'll get everything as clean as possible, try a few more games, and report back.

**Comment by Timmeey86** [comment]
@cpinter @TheHandiCraftsman The bugs you describe is exactly what the CQUI Summer version behaved like after the Fall update. Are you sure you didn't download the master by accident?

**Comment by oerms** [comment]
@cpinter @TheHandiCraftsman Also, rename the extracted folder: delete "fall-2017".

**Comment by LordYanaek** [comment]
@Timmeey86 & @Azurency sorry for the false report then, i checked DebugLocalization.sqlite and LOC_GOSSIP_SOURCE_DELEGATE is indeed empty for en_US only. That would also explain why i occasionally had missing localization (such as on the change policy reminder popup) and maybe why i had some German text occasionally show up with the previous version (i considered it a minor glitch).

Took me some time to find something wrong. I had a "ghost" of an old chaorace CQUI (UUID : 03B8BDDE-DEAD-BEEF-9540-26E685E3156E) in the content browser inside Civ6 but couldn't find it in either mod folders (workshop or regular) and kept it disabled. Following it's tracks in Mods.sqlite i finally found it inside my dlc folder. Turns out i tried to get it working as a DLC like EUI did in Civ5 in order to be able to use it with saved games created without CQUI (before the AffectsSavedGames tag was introduced/discovered) and then forgot it was there.

Unfortunately that didn't solve the issue. LOC_GOSSIP_SOURCE_DELEGATE is still empty and the t77 save still locks with no UI. I already verified my files with Steam. I'll try a full uninstall and reinstall.

**Comment by OldFartAtPlay** [comment]
Just created a github account so that I could log in and add my thanks to you all. CQUI improves Civ VI by an order of magnitude, so what you are doing really makes a difference.
Incidentally, as a retired software developer, I'm fascinated by github - it looks like an excellent system!
Thanks again, and keep up the good work.

**Comment by vtmatt** [comment]
I haven't seen this one mentioned yet (though I have a 6 day old newborn baby keeping me up, so it's possible I just missed it in my sleep deprived state!).  

I started a full earth game with all civs except one (only one Greek leader) in play.  When I check the score tab on world rankings, it displays the "You are in xxxxxx place" message wrong.   Appears to be only if you are 21st or worse, though that's anecdotal, as I haven't seen every potential ranking yet.  

Image attached.  Not sure off the top of my head how to do log/save-game upload, but if either are needed, let me know and I'll add those ASAP.

![cquiranking](https://user-images.githubusercontent.com/16530035/32404837-30f643fc-c12f-11e7-832f-1f495565390d.png)

**Comment by georgwacker** [comment]
**Re Promotion-Bug**: I tested with your recent commit and also astogs PR for the lens-fix. With the lens-fix it now shows the promotion icon on turn 14 when I load the game, but after moving the scout once, it disappears again (icon and button) on the next turn, even though the promotion is still available. A couple of turns later it shows up again.

[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1443202/Logs.zip)
[TEDDY ROOSEVELT 14 3480 BC.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1443195/TEDDY.ROOSEVELT.14.3480.BC.zip)

**Comment by oerms** [comment]
@vtmatt 
**Ranking with variable name** The game itself does not support rankings for more than 20 players, as seen in `%steamapps&\common\Sid Meier's Civilization VI\Base\Assets\Text\en_US\InGameText.xml` lines 3952 and following. I try and generate a file for this to go up to 30. Do you think this is enough?

**Policy Reminder** This remainder is intended to remind you to set newly available policies before you accidentally end the turn. It costs money to do it the next turn. We should implement that it stays off after you discarded it the first time during the turn.

**Comment by skodkim** [comment]
Small issue with the world builder using the beta from a few hours ago: Assigned starting positions on a map are invisible on the map (but clicking in the time you can se that it is still there).

\Skodkim

**Comment by Azurency** [comment]
@vtmatt That's funny, Larkana is an actual city close to Mohenjo-Daro in Pakistan, I think that's why the dev changed the name to that, I don't think it has anything to do with CQUI.
For the policy reminder popup, it's an issue that were already here before the patch (but the policy reminder popup was never pushed to steam) and I forgot to fix it, it's done now.

@georgwacker I have tested with your save, it's working fine. Is the promote button replaced by the cancel button ? I think it's a normal behavior, you can't consume all your movements AND upgrade the unit. Other than that, for me it's always showing fine if I have movements left.

@skodkim Is CQUI needed in the world builder ?

**Comment by Azurency** [comment]
I think we're ready to merge everything back to master, CQUI seems to work well and there's no game breaking problems. I'll do that this evening, then @sparrow8332 will update the steam version. The remaining bugs, if any, will then have their own issue.

**Comment by imkailasraj** [comment]
i don't know if this issue is addressed or not.the top bat(faith gold etc)flickers after pressing the next turn.And also how to update when new version comes?will the existing game be affected?

**Comment by Azurency** [comment]
@georgwacker oh I think I see what you mean. You're talking about the flag on the unit, the red + sign, not the action button on the unit panel ? Well yes there's a bug here I just noticed, it's not showing when a unit have an order to move somewhere but can have a promotion. It's because CQUI check for `CanStartCommand` but it return false if there is no movement left, I'll fix this.

@imkailasraj you're using the summer version of CQUI not the fall-2017 version you can find on the appropriate branch here on GitHub. If you're subscribe on steam, just wait for the update.

**Comment by georgwacker** [comment]
Turn 14 loaded:
![promotion_correct](https://user-images.githubusercontent.com/13589801/32407255-990a7e2a-c186-11e7-9e91-da26c0bfee29.PNG)

Next turn:
![promotion_nextturn](https://user-images.githubusercontent.com/13589801/32407259-afca20d4-c186-11e7-92d8-67112bcc19e3.PNG)

Both disappear. But the little icon on the unit also vanishes with vanilla (just loaded the save there).

**Comment by sparrow8332** [comment]
@georgwacker have you tried the latest fall-2017 branch ? im sure @Azurency fixed this but i could be wrong

**Comment by skodkim** [comment]
@Azurency AFAIK cqui is not needed at all when you use the world builder. 

It's still loaded though (small tobacco icon work cqui options visible) and somehow it causes starting positions to become invisible. After removing the mod they came back again. It's not a priority for now I guess.

Keep up the good work - it's amazing what you doing!

\Skodkim

**Comment by Azurency** [comment]
@georgwacker @sparrow8332 It's not a bug 🙈 you just can promote a unit when it have not move left, and giving it an order to move to somewhere consume all the movements it can, but is played at the end of the turn, it's the "1" you see in the screenshot. To promote you can to cancel the order. 
For the flag, I'm trying to fix it now, by comparing the current exp and the max exp and always show the "+" sign if it >=

**Comment by cpinter** [comment]
@Timmeey86 @oerms I'm absolutely sure I got the proper branch. I also renamed the folder. But I didn't unsubscribe from it in steam just disabled that mod, and enabled the other one (the new one that appeared once I copied it over). It does seem that the old mod is used, and not sure what goes wrong. Anyway, it seems I cannot be a lot of help with testing like this...

**Comment by Azurency** [comment]
**Last update**
Well, I think we're all good now. Thank you ,everyone, for the help, contributions and bug finding 👍 we would not have been so fast without all that. 
I've merged the update to the master branch 🎉 and the Steam version will very soon be updated.

If you find a bug, please open a new issue and not post it here.

Back to playing and implementing new stuff then.

**Comment by lanych** [comment]
Great job, guys!! You did it!
The civ is shining now again ))

**Comment by orochi28** [comment]
![policy change reminder bug](https://user-images.githubusercontent.com/33386307/32411984-c64c32bc-c1b7-11e7-86da-0c9a8b236ca8.jpg)

Thank you guys so much for the timely update! I'm excited to play with CQUI again.

Looks like the latest update included a policy change reminder? I somehow got it on turn 1, when there was no government or policy to choose from. When I click on "Change Government", nothing happens and the dialog box stays there, though I am able to close the dialog and end turn. Wondering if it's a bug?

**Comment by Azurency** [comment]
@orochi28 Yes that's not supposed to happen, it should be hidden upon initialization. If you still have the issue, can you open a new issue here on Github ? I'll close this one.
